.github/workflows/twisted_trunk.yml

```
@@ -96,7 +96,7 @@ jobs:
     if: needs.check_repo.outputs.should_run_workflow == 'true'
     runs-on: ubuntu-latest
     container:
-      image: matrixdotorg/sytest-synapse:buster
+      image: matrixdotorg/sytest-synapse:focal
```

It looks like we're switching from a Debian version name to a Ubuntu one. Is that expected?

I assume we want to use `debian:bullseye` next? Or if we don't care about using the oldest version, using whatever is the LTS nowadays

---

Ahh, I didn't properly check this. [bullseye is using Python 3.9](https://packages.debian.org/bullseye/python3) whereas focal is on 3.8.

---

If running on the oldest Python version matters here, we should add some comments why:

```suggestion
      # We're using ubuntu:focal because it uses Python 3.8 which is our minimum supported Python version
      # (bonus points for why  minimum supported Python version matters)
      image: matrixdotorg/sytest-synapse:focal
```

---

I asked about this and didn't get clarity as to *why* we use the oldest, would you be happy with the comment you suggested minus the bonus points part?

---

Perhaps this then:

```suggestion
      # We're using ubuntu:focal because it uses Python 3.8 which is our minimum supported Python version.
      # There is no particular reason for why we use the minimum supported Python version beyond that it ensures compatibility there.
      image: matrixdotorg/sytest-synapse:focal
```

---

FWIW, my headcanon here: this job is a canary to warn us about unreleased twisted changes that would cause problems for us if they were to be released immediately. We could do this by testing against all Python versions, but for simplicity's sake (and to save CI runners) we use the oldest version, assuming that any incompatibilities on newer versions would also be present on the oldest.

---

Adding that context sounds great to me :+1: